src/app/components/character-count/index.html

```diff
@@ -0,0 +1,284 @@
+
+        <sky-docs-visual-example>
+          <sky-docs-visual-example-text>
+            <p>
+              Don’t use character count on inputs when exceeding the character limit is unrealistic. Use normal form validation to handle errors caused by character limit violations.
+            </p>
+          </sky-docs-visual-example-text>
+
+          <sky-docs-visual-example-image>
+            <div
+              class="sky-margin-stacked-content-section"
+            >
+              <img
+                src="~/assets/img/guidelines/character-count/character-count-usage-3.png"
+              />
+              <div
+                class="caption caption-danger sky-text-danger"
+              >
+                Don't use character count when the character limit is unlikely to be exceeded.
+              </div>
+            </div>
+
+            <div>
+              <img
+                src="~/assets/img/guidelines/character-count/character-count-usage-4.png"
+              />
+              <div
+                class="caption caption-success sky-text-success"
+              >
+                Do use normal form validation for edge cases where a high character limit is exceeded.
+              </div>
+            </div>
+          </sky-docs-visual-example-image>
+        </sky-docs-visual-example>
+      </sky-demo-page-content>
+
+      <sky-demo-page-content sectionHeading="Anatomy">
+
+        <sky-docs-anatomy
+          imagePath="~/assets/img/guidelines/character-count/character-count-anatomy-1.png"
+        >
+          <sky-docs-anatomy-item
+            anatomyNumber="1"
+          >
+            Input field
+          </sky-docs-anatomy-item>
+          <sky-docs-anatomy-item
+            anatomyNumber="2"
+          >
+            Character count/limit
+          </sky-docs-anatomy-item>
+          <sky-docs-anatomy-item
+            anatomyNumber="3"
+          >
+            Danger icon
+          </sky-docs-anatomy-item>
+          <sky-docs-anatomy-item
+            anatomyNumber="4"
+          >
+            <a routerLink="/components/status-indicator"> Status indicator</a> (danger)
+          </sky-docs-anatomy-item>
+        </sky-docs-anatomy>
+      </sky-demo-page-content>
+
+      <sky-demo-page-content
+        sectionHeading="Behavior & states"
+      >
+        <div
+          class="sky-text-block"
+        >
+          <p>
+            The character count component is an extension of the <a routerLink="/components/form">form</a> component, which defines the styles and appearance of the input field states.
+          </p>
+        </div>
+
+        <sky-demo-page-component-states>
+          <sky-demo-page-component-state
+            stateName="Empty field"
+            imagePath="~/assets/img/guidelines/character-count/character-count-state-1.png"
+          >
+          </sky-demo-page-component-state>
+          <sky-demo-page-component-state
+            stateName="Adding text"
```

Comment:
I like the way that the entries walk through adding text, but are the `Adding text` and `At character limit` items really distinct states? I don't think so, but I'm not sure if it matters either way. Just asking to make sure that if they aren't actually distinct states that we are making a conscious decision to present them as such in order to illustrate how the component works.

Reply:
Just a conscious decision to show that there's no difference in state when the character limit is matched exactly. It's only when the limit is exceeded that there's a state change. Yes, technically, that means they are not "two different states", but I believe two different examples are needed to make the point completely clear.